src/main/scala/com/cognite/sdk/scala/common/K8sServiceToken.scala

```diff
@@ -0,0 +1,27 @@
+package com.cognite.sdk.scala.common
+
+import cats.effect.Async
+
+import java.nio.charset.Charset
+import java.nio.file.{Files, Paths}
+import scala.util.Try
+
+/** Only use for SessionProvider to interact with Cognite internal SessionAPI */
+trait K8sServiceTokenProvider {
+  def getKubernetesJwt[F[_]](implicit F: Async[F]): F[String]
+}
+
+object K8sServiceToken extends K8sServiceTokenProvider {
+  def getKubernetesJwt[F[_]](implicit F: Async[F]): F[String] = {
+    val serviceAccountTokenPath = "/var/run/secrets/tokens/cdf_token"
+    Try(
+      new String(Files.readAllBytes(Paths.get(serviceAccountTokenPath)), Charset.defaultCharset)
+    ).toEither match {
+      case Right(token) => F.pure(token)
+      case Left(err) =>
+        F.raiseError(
+          new SdkException(s"Failed to get service token because ${err.getMessage}")
```

Comment:
I'm not sure this is a good idea for this case, but I don't know. It's more of an internal error if we get to this point, I think?